src/parameterizations/vertical/MOM_opacity.F90

```diff
@@ -451,41 +451,55 @@ function opacity_manizza(chl_data)
 
 !> This subroutine returns a 2-d slice at constant j of fields from an optics_type, with the potential
 !! for rescaling these fields.
-subroutine extract_optics_slice(optics, j, G, GV, opacity, opacity_scale, penSW_top, penSW_scale)
+subroutine extract_optics_slice(optics, j, G, GV, opacity, opacity_scale, penSW_top, penSW_scale, SpV_avg)
   type(optics_type),       intent(in)  :: optics !< An optics structure that has values of opacities
                                                  !! and shortwave fluxes.
   integer,                 intent(in)  :: j      !< j-index to extract
   type(ocean_grid_type),   intent(in)  :: G      !< The ocean's grid structure.
   type(verticalGrid_type), intent(in)  :: GV     !< The ocean's vertical grid structure.
   real, dimension(max(optics%nbands,1),SZI_(G),SZK_(GV)), &
                  optional, intent(out) :: opacity   !< The opacity in each band, i-point, and layer [Z-1 ~> m-1],
-                                                    !! but with units that can be altered by opacity_scale.
+                                                    !! but with units that can be altered by opacity_scale
+                                                    !! and the presence of SpV_avg to change this to other
+                                                    !! units like [H-1 ~> m-1 or m2 kg-1]
   real,          optional, intent(in)  :: opacity_scale !< A factor by which to rescale the opacity [nondim] or
                                                     !! [Z H-1 ~> 1 or m3 kg-1]
   real, dimension(max(optics%nbands,1),SZI_(G)), &
                  optional, intent(out) :: penSW_top !< The shortwave radiation [Q R Z T-1 ~> W m-2]
                                                     !! at the surface in each of the nbands bands
                                                     !! that penetrates beyond the surface skin layer.
-  real,          optional, intent(in)  :: penSW_scale !< A factor by which to rescale the shortwave flux [nondim]?
+  real,          optional, intent(in)  :: penSW_scale !< A factor by which to rescale the shortwave flux [nondim]
+                                                    !! or other units.
```

Comment:
"other units" should be H_to_RZ -> kg m-3

Reply:
This optional argument to `extract_optics_slice()` is never actually used, so there is no way to determine what other units it might actually be used for.  I think that the two viable options are to leave the vague unit description that is presently in this PR (the safer option?), or we could simply eliminate the optional `penSW_scale` argument altogether.  Which option would you prefer, @breichl?

Reply:
Thanks for this explanation.  I'm fine w/ keeping as is.